workshops/discord_chatbot/README.md

```diff
@@ -0,0 +1,372 @@
+
+![name agent](https://cloud-9x2pb85rb.vercel.app/screenshot8.png)
+
+Now hit the Create button and wait for a few seconds for the agent to be created.
+
+After a while, you will be greeted with this screen.
+
+![create bot](https://cloud-9x2pb85rb.vercel.app/screenshot9.png)
+
+Now click on the "Small Talk" tab on the menu on the left side, and go ahead and enable the small talk toggle. This is essential if you want your bot to be able to converse with you.
+
+![small talk](https://cloud-9x2pb85rb.vercel.app/screenshot10.png)
+
+Now, we need to get API keys from dialogflow just like how we did it with Discord. For this, click on the gear icon beside your agent name. This will take you to the settings pannel of agent.
+Here click on your `Project ID` and open it in a new tab.
+
+![agent settings](https://cloud-q4uz3wse3.vercel.app/screenshot15.png)
+
+Now go to `Project settings` in your GCP console. If you haven't logged in to GCP, make sure you are using the same Google account with which you have created the Dialogflow agent with.
+
+![gcp console](https://cloud-q4uz3wse3.vercel.app/screenshot16.png)
+
+Now, you will be greeted with this page. Go to `Service Accounts` on the left hand menu.
+
+![gcp service account creation](https://cloud-q4uz3wse3.vercel.app/screenshot17.png)
+
+Now, click on the link for the `Service Account`.
+
+![GCP service account](https://cloud-cl244cmpv.vercel.app/screenshot12.png)
+
+Here, click on your service account down there.
+(If your service doesn't exist, then wait for a while for it to get created. If it still doesn't appear, then create the agent once again. If this too doesn't work, then go ahead and create a new service account using the button beside `Service Accounts`)
+
+![add key](https://cloud-cl244cmpv.vercel.app/screenshot13.png)
+
+Now, click on "Add Key" and create a new key in JSON format.
+
+(Create a new key even if a previous one exists)
+
+This should automatically get downloaded for you to your local system.
+
+This is your API key from Dialogflow. Keep this safe, as it has private information about your dialogflow account.
+
+Now that you have come so far, give yourself a pat on your back, because we are all set up to start coding the bot using discord.js and dialogflow APIs.
+
+![pat on back](https://media.giphy.com/media/9Q249Qsl5cfLi/giphy.gif)
+
+## Part 2: Start coding the bot!
+
+Here, we will be using JavaScript as a programming language to code out the discord bot. I will be using code snippets from [discord.js](https://discord.js.org/) and [Node.js Dialogflow client](https://github.com/googleapis/nodejs-dialogflow#using-the-client-library) documentation. So if you're interested you can have a read before we move on!
+
+Now let's get our hands dirty and start coding the bot!
+
+Go ahead to [this repl link](https://repl.it/@k4u5h4L/discord-chatbot) to code out the bot online! All the packages are already installed for you there so that you can directly dive into the code!
```

Comment:
This paragraph and the next one can be shortened to "Fork the [starter project](https://repl.it/@k4u5h4L/discord-chatbot) to start coding."